### PR TITLE
Paul posting stuff

### PR DESCRIPTION
**Pull request by paul80**
Can display public posts, posts by/to friends (if not private); doesn't look nice of course but can be fixed later on. Has custom middleware to prevent access to urls a user shouldn't be able to access. Don't know if it is needed or not though.

### CODE REVIEW

**Comment by klindenbach** [comment]
Can you maybe explain a little bit about the middleware? It looks like it redirects you to the login page if not authenticated, with the exception of `/admin`  and `/register`.

Is that correct?

**Comment by paul80** [comment]
Yeah that's the gist of it. Without it I suppose we could just use the @login_required decorator for every view. Up to you guys if you want it or not.

**Comment by klindenbach** [comment]
Does `/login` have to be login exempt?
Fix your merge conflicts and you've got the :+1: from me.

File: socialdistribution/urls.py [hunk 1]
```diff
@@ -12,7 +12,8 @@
     # url(r'^blog/', include('blog.urls')),
 
     url(r'^admin/', include(admin.site.urls)),
-    url(r'^friends/', include('author.urls')),
+    #Gives an error when uncommented likely because there is no urls.py in author app 
+    #url(r'^friends/', include('author.urls')),
```
Comment (klindenbach) [inline socialdistribution/urls.py]: Can we take out these changes now?

Comment (paul80) [inline socialdistribution/urls.py]: yeah ok i will take em out

Comment (paul80) [inline socialdistribution/urls.py]: No, don't thijnk /login has to be login exempt. Was trying it out yesterday and didn't encounter a problem with it

File: socialdistribution/urls.py [hunk 2]
```diff
@@ -12,7 +12,6 @@
     # url(r'^blog/', include('blog.urls')),
 
     url(r'^admin/', include(admin.site.urls)),
-    url(r'^friends/', include('author.urls')),
```
Comment (klindenbach) [inline socialdistribution/urls.py:15]: Still took it out

Comment (paul80) [inline socialdistribution/urls.py:15]: oh whoops ok will readd that part

**Pull request merged**